integrations/sendgrid/src/index.ts

```diff
@@ -0,0 +1,80 @@
+import { IntegrationClient, TriggerIntegration } from "@trigger.dev/sdk";
+import sgMail from "@sendgrid/mail";
+
+import type { AuthenticatedTask } from "@trigger.dev/sdk";
+
+type SendEmailData = Parameters<typeof sgMail["send"]>[0];
+type SendEmailResponse = [sgMail.ClientResponse, {}];
+
+
+export const sendEmail: AuthenticatedTask<typeof sgMail, SendEmailData, SendEmailResponse> = {
+  run: async (params, client) => {
+    const response = await client.send(params);
+    return response as SendEmailResponse; 
+  },
+  init: (params) => {
+    const subjectProperty = Array.isArray(params) ? [] : (params.subject ? [{ label: "Subject", text: params.subject }] : []);
+    
+    return {
+      name: "Send Email",
+      params,
+      icon: "sendgrid", 
+      properties: [
+        {
+          label: "From",
+          text: Array.isArray(params) ? params[0].from : params.from, 
```

Comment:
I got TypeScript errors in VSCode that were complaining about the `init` function.

It's because `.from` and `.to` are the EmailData type which is of this type:

```ts
type EmailData = string | { name?: string; email: string };
```
You could add a function like this to the end of this file which gets a string from from/to

```ts
type EmailData = string | { name?: string; email: string };

function getEmailFromEmailData(emailData: EmailData | undefined): string {
  if (emailData === undefined) {
    return "";
  }

  if (typeof emailData === "string") {
    return emailData;
  }
  return emailData.email;
}
```

You would use it like this:

```ts
text: Array.isArray(params)
            ? getEmailFromEmailData(params[0].from)
            : getEmailFromEmailData(params.from),
```

Reply:
Thanks, I've fixed this